package/gluon-mesh-batman-adv/src/respondd.c

```diff
@@ -76,49 +81,63 @@ struct clients_netlink_opts {
 	struct batadv_nlquery_opts query_opts;
 };
 
+static int get_addresses_cb(struct nl_msg *msg, void *arg) {
+    struct nlmsghdr *nlh = nlmsg_hdr(msg);
+    struct ifaddrmsg *msg_content = NLMSG_DATA(nlh);
+    struct rtattr *hdr = IFA_RTA(msg_content);
+    int remaining = nlh->nlmsg_len - NLMSG_LENGTH(sizeof(struct ifaddrmsg));
 
-static struct json_object * get_addresses(void) {
-	FILE *f = fopen("/proc/net/if_inet6", "r");
-	if (!f)
-		return NULL;
+    int br_client_ifindex = if_nametoindex("br-client");
```

Comment:
`if_nametoindex` should only be called once and passed into the callback (by making arg a struct containing the ifindex and the JSON array).

Reply:
Good idea, will implement this.